server/channels/app/channel.go

```diff
@@ -3703,6 +3771,15 @@ func (a *App) validateForConvertGroupMessageToChannel(c request.CTX, convertedBy
 		}
 	}
 
+	// we can't convert GM to a channel if the team is not common among all GM members
+	// since we are making this automagically, we are taking the first team from the common teams
+	if gmConversionRequest.TeamID == "" && len(commonTeams) > 0 {
+		gmConversionRequest.TeamID = commonTeams[0].Id
+		teamFound = true
+	} else if len(commonTeams) == 0 {
+		return model.NewAppError("validateForConvertGroupMessageToChannel", "app.channel.group_message_conversion.no_common_teams", nil, "", http.StatusBadRequest)
+	}
```

Comment:
Is this check needed? If there are no common teams, then the parent if would not be true, so `teamFound` will be anyways false. And then it would get returned in the next if condition.

Reply:
Yes because we are handling this error separately on the call site (for app migration and permanent deletion case). This is also being used from the webapp so users may handle it differently.